### PR TITLE
Adding configuration hooks for each section of the haproxy.cfg through attribute hashes

### PR DESCRIPTION
**Pull request by bbergstrom**
Greater flexibility for custom configuration without duplicating code in consuming cookbooks.

Depends on https://github.com/sportngin-cookbooks/opsworks-haproxy/pull/1

### CODE REVIEW

File: templates/default/haproxy.cfg.erb [hunk 1]
```diff
@@ -1,67 +1,85 @@
...
   option redispatch
   option httpclose
   option forwardfor
   option httpchk /okcomputer
+  <% @custom_config[:backend].each do |name, value| %>
+  <%= "#{name} #{value}" unless value.nil? %>
+  <% end %>
   <% @backends.each do |name, attrs| -%>
-    server <%= name %> <%= attrs['private_ip'] %>:81 check port 81 inter 5000 fastinter 1000 fall 1 weight <%= attrs[:weight] || 0 %>
+  server <%= name %> <%= attrs['private_ip'] %>:81 check port 81 inter 5000 fastinter 1000 fall 1 weight <%= attrs[:weight] || 0 %>
```
Comment (anfleene) [inline templates/default/haproxy.cfg.erb]: Port should probably be an attribute

File: templates/default/haproxy.cfg.erb [hunk 2]
```diff
@@ -1,67 +1,85 @@
...
   <% end -%>
 <% end -%>
 
 <% unless @backends.empty? -%>
-  backend app_servers_ssl
+backend app_servers_ssl
   mode tcp
   balance leastconn
   option redispatch
   option httpchk /okcomputer
```
Comment (anfleene) [inline templates/default/haproxy.cfg.erb]: the health check should be an attribute

**Comment by bbergstrom** [comment]
@anfleene Revised

File: templates/default/haproxy.cfg.erb [hunk 3]
```diff
@@ -1,67 +1,91 @@
...
+  option httpchk <%= node[:haproxy][:health_check_method] %> <%= node[:haproxy][:health_check_url] %>
+
+  <% @custom_config[:backend].each do |name, value| %>
+  <%= "#{name} #{value}" unless value.nil? %>
+  <% end %>
+
+  default-server <%= OpsworksHaproxy::default_server_params(node) %>
   <% @backends.each do |name, attrs| -%>
-    server <%= name %> <%= attrs['private_ip'] %>:81 check port 81 inter 5000 fastinter 1000 fall 1 weight <%= attrs[:weight] || 0 %>
+  server <%= "#{name} #{attrs['private_ip']}:#{node[:haproxy][:backend_port]} check weight #{attrs[:weight]}" %>
```
Comment (anfleene) [inline templates/default/haproxy.cfg.erb]: how are you handling all the other options here?

Comment (bbergstrom) [inline templates/default/haproxy.cfg.erb]: port, inter, fastinter and fall are being set in the default server line which applys to all servers in the current backend.

Comment (anfleene) [inline templates/default/haproxy.cfg.erb]: Nice!

**Comment by anfleene** [comment]
These changes look great!

**Pull request merged**